pennylane_lightning/lightning_qubit.py

```diff
@@ -342,6 +365,37 @@ def _apply_operation():
     def _apply_unitary():
         pass
 
+    @staticmethod
+    def operations_batch_size(operations):
+        """Go through the operations list to find the batch size.
+        It is assumed that after some preprocessing the batch_size for each operation will be None or a same number.
+
+        Args:
+            operations (list[~pennylane.operation.Operation]): list of operation
+
+        Returns:
+            int: the batch size for operators.
+        """
+        for o in operations:
+            if o.batch_size:  # None is False for python.
+                return o.batch_size  # The search will stop when finding the first no None entry.
+        return None
+
+    def check_and_guarantee_batched_states_shape(self, state):
+        """Check if the state is batched and if not batch it with a batched dimension equal 1.
+
+        Args:
+            state (array[complex]): the input state tensor
+
+        Returns:
+            The batch dimension (None, if no batching is in action.),
+            and the state with an extra dimension for processing if not batching.
+        """
+        dim = 1 << self.num_wires
+        batch_size = self._get_batch_size(state, (dim,) * self.num_wires, dim)
```

Comment:
I'm not sure I understand the `expected_shape` `(dim,)*self.num_wires` here. Why should the state on a device with 3 wires have shape `(8, 8, 8)`?

Reply:
Oops. Thank you for catching that.